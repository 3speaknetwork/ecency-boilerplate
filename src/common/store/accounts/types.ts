import { ExtendedAccount } from "@esteemapp/dhive";

export interface Account extends ExtendedAccount {}

export interface State {
  data: Account;
  loading: boolean;
  error: boolean;
}

export enum ActionTypes {
  FETCH = "@account/FETCH",
  FETCHED = "@account/FETCHED",
  FETCH_ERROR = "@account/FETCH_ERROR"
}

export interface FetchAction {
  type: ActionTypes.FETCH;
}

export interface FetchedAction {
  type: ActionTypes.FETCHED;
  data: Account;
}

export interface FetchErrorAction {
  type: ActionTypes.FETCH_ERROR;
}

export type Actions = FetchAction | FetchedAction | FetchErrorAction;