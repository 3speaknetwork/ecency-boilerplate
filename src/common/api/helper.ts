import isElectron from "../util/is-electron";

import defaults from "../constants/defaults.json";

export const apiBase = (endpoint: string): string => {
    let base = '';

    if (isElectron()) {
        if (process.env.NODE_ENV === "development") {
            base = "http://localhost:3000";
        } else {
            base = defaults.base;
        }
    }

    return `${base}${endpoint}`;
}
