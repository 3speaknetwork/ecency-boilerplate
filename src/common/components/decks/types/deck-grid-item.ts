interface BaseDeckGridItem {
  key: number;
  type: "u" | "ac" | "co" | "w" | "n" | "tr" | "to" | "s" | "cu";
  settings: unknown;
}

export interface UserDeckGridItem extends BaseDeckGridItem {
  type: "u";
  settings: {
    username: string;
    contentType: string;
  };
}

export type DeckGridItem = BaseDeckGridItem | UserDeckGridItem;
