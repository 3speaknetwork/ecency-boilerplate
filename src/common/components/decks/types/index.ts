export * from "./deck-grid-item";
export * from "./deck-grid";
