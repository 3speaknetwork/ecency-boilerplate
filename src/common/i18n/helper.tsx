import React from "react";

import {_t} from "./index";

interface Props {
    k: string,
    args: {},
    children: JSX.Element
}

export class Tsx extends React.Component<Props> {
    render() {
        const {k, args} = this.props;
        const {children} = this.props;
        return React.cloneElement(children, {dangerouslySetInnerHTML: {__html: _t(k, args)}});
    }
}
