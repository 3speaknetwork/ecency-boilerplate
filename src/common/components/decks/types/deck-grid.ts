import { DeckGridItem } from "./deck-grid-item";

export interface DeckGrid {
  columns: DeckGridItem[];
}
