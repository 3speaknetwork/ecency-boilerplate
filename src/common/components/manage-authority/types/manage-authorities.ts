import { Account } from "../../../store/accounts/types";

export const Keytype = {
  Owner: "owner",
  Active: "active",
  Posting: "posting"
};

export interface PublicKeys {
  publicOwnerKey?: any;
  publicActiveKey?: any;
  publicPostingKey?: any;
}

export interface AccountDataType {
  postingsAuthority: Array<any>;
  posting: Array<any>;
  owner: Array<any>;
  active: Array<any>;
  weight: number;
  memokey: string;
  account: Account;
  PublicKeys: {
    publicOwnerKey: any;
    publicActiveKey: any;
    publicPostingKey: any;
  };
}

export const actionType = {
  Revoke: "revoke"
};
