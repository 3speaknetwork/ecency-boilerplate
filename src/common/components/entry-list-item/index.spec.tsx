import React from "react";

import {createBrowserHistory, createLocation} from "history";

import mockDate from "mockdate";

import {StaticRouter} from "react-router-dom";

import TestRenderer from "react-test-renderer";

import {globalInstance, dynamicPropsIntance1, entryInstance1, UiInstance, activeUserMaker} from "../../helper/test-helper";

import {ListStyle} from "../../store/global/types";

import EntryListItem from "./index";


mockDate.set(1591398131174);

const defProps = {
    history: createBrowserHistory(),
    location: createLocation({}),
    global: globalInstance,
    dynamicProps: dynamicPropsIntance1,
    communities: [],
    community: null,
    users: [],
    activeUser: null,
    reblogs: [],
    entry: entryInstance1,
    ui: UiInstance,
    asAuthor: "",
    promoted: false,

    addAccount: () => {
    },
    updateEntry: () => {
    },
    setActiveUser: () => {
    },
    updateActiveUser: () => {
    },
    deleteUser: () => {
    },
    addReblog: () => {
    },
    toggleUIProp: () => {
    },
}

it("(1) Default render", () => {
    const renderer = TestRenderer.create(
        <StaticRouter location="/" context={{}}>
            <EntryListItem {...defProps} />
        </StaticRouter>);
    expect(renderer.toJSON()).toMatchSnapshot();
});

it("(2) Grid view", () => {

    const props = {
        ...defProps,
        global: {
            ...globalInstance,
            listStyle: ListStyle.grid
        }
    }
    const renderer = TestRenderer.create(
        <StaticRouter location="/" context={{}}>
            <EntryListItem {...props} />
        </StaticRouter>);
    expect(renderer.toJSON()).toMatchSnapshot();
});


it("(3) Nsfw", () => {
    const props = {
        ...defProps,
        entry: {
            ...entryInstance1,
            json_metadata: {
                ...entryInstance1.json_metadata,
                tags: [...entryInstance1.json_metadata.tags, "nsfw"]
            }
        }
    }
    const renderer = TestRenderer.create(
        <StaticRouter location="/" context={{}}>
            <EntryListItem {...props} />
        </StaticRouter>);
    expect(renderer.toJSON()).toMatchSnapshot();
});


it("(4) Nsfw with active user", () => {
    const props = {
        ...defProps,
        entry: {
            ...entryInstance1,
            json_metadata: {
                ...entryInstance1.json_metadata,
                tags: [...entryInstance1.json_metadata.tags, "nsfw"]
            }
        },
        activeUser: activeUserMaker("foo")
    }
    const renderer = TestRenderer.create(
        <StaticRouter location="/" context={{}}>
            <EntryListItem {...props} />
        </StaticRouter>);
    expect(renderer.toJSON()).toMatchSnapshot();
});

it("(5) Nsfw but allowed", () => {
    const props = {
        ...defProps,
        entry: {
            ...entryInstance1,
            json_metadata: {
                ...entryInstance1.json_metadata,
                tags: [...entryInstance1.json_metadata.tags, "nsfw"]
            }
        },
        global: {
            ...globalInstance,
            nsfw: true
        }
    }
    const renderer = TestRenderer.create(
        <StaticRouter location="/" context={{}}>
            <EntryListItem {...props} />
        </StaticRouter>);
    expect(renderer.toJSON()).toMatchSnapshot();
});
