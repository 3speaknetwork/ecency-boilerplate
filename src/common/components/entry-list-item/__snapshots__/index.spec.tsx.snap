// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Default render 1`] = `
<div
  className="entry-list-item"
>
  <div
    className="item-header"
  >
    <div
      className="item-header-main"
    >
      <div
        className="author-part"
      >
        <a
          className="author-avatar"
          href="/@good-karma"
          onClick={[Function]}
        >
          <span
            className="user-avatar small"
            style={
              Object {
                "backgroundImage": "url(https://images.ecency.com/u/good-karma/avatar/small)",
              }
            }
          />
        </a>
        <a
          className="author notranslate"
          href="/@good-karma"
          onClick={[Function]}
        >
          good-karma
          <span
            className="author-reputation"
          >
            76
          </span>
        </a>
      </div>
      <a
        className="category"
        href="/hot/hive"
        onClick={[Function]}
      >
        hive
      </a>
      <span
        className="read-mark"
      />
      <span
        className="date"
        title="Monday, June 1, 2020 8:53 AM"
      >
        5 days
      </span>
    </div>
    <div
      className="item-header-features"
    />
  </div>
  <div
    className="item-body"
  >
    <div
      className="item-image"
    >
      <a
        href="/hive/@good-karma/awesome-hive"
        onClick={[Function]}
      >
        <picture>
          <source
            media="(min-width: 576px)"
            srcSet="https://images.ecency.com/p/HNWT6DgoBc1692QWn5trsGLxi44TrGuNZJMEdibBBbUKuho3EaW43JV4QzaQ258PEGNV2u43KAhU8r3Vjno4cBP8JeTFkbS2SCMjZZPprKzfMSdteRxSZxuPku4?format=match&mode=fit&width=260&height=200"
          />
          <img
            alt="Awesome Hive"
            onError={[Function]}
            srcSet="https://images.ecency.com/p/HNWT6DgoBc1692QWn5trsGLxi44TrGuNZJMEdibBBbUKuho3EaW43JV4QzaQ258PEGNV2u43KAhU8r3Vjno4cBP8JeTFkbS2SCMjZZPprKzfMSdteRxSZxuPku4?format=match&mode=fit&width=600&height=500"
          />
        </picture>
      </a>
    </div>
    <div
      className="item-summary"
    >
      <a
        className="item-title"
        href="/hive/@good-karma/awesome-hive"
        onClick={[Function]}
      >
        Awesome Hive
      </a>
      <a
        className="item-body"
        href="/hive/@good-karma/awesome-hive"
        onClick={[Function]}
      >
        Hey developers, I just created Awesome-Hive, list of services and apps on Hive. If you don't know about Awesome project, take a look at here: Project contains list of services and links to various too
      </a>
    </div>
    <div
      className="item-controls"
    >
      <div
        className="entry-vote-btn"
        onClick={[Function]}
      >
        <div
          className="btn-vote btn-up-vote"
        >
          <span
            className="btn-inner"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="entry-payout  notranslate"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        $ 125.10
      </div>
      <div
        className="entry-votes notranslate"
      >
        <span
          className="inner-btn"
          onClick={[Function]}
          title="10 Votes"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M16 17V19H2V17S2 13 9 13 16 17 16 17M12.5 7.5A3.5 3.5 0 1 0 9 11A3.5 3.5 0 0 0 12.5 7.5M15.94 13A5.32 5.32 0 0 1 18 17V19H22V17S22 13.37 15.94 13M15 4A3.39 3.39 0 0 0 13.07 4.59A5 5 0 0 1 13.07 10.41A3.39 3.39 0 0 0 15 11A3.5 3.5 0 0 0 15 4Z"
              fill="currentColor"
            />
          </svg>
           
          10
        </span>
      </div>
      <a
        className="replies notranslate"
        href="/hive/@good-karma/awesome-hive"
        onClick={[Function]}
      >
        <span
          className="inner"
          title="31 responses. Click to respond."
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M18,8H6V6H18V8M18,11H6V9H18V11M18,14H6V12H18V14M22,4A2,2 0 0,0 20,2H4A2,2 0 0,0 2,4V16A2,2 0 0,0 4,18H18L22,22V4Z"
              fill="currentColor"
            />
          </svg>
           
          31
        </span>
      </a>
      <div
        className="entry-reblog-btn"
        onClick={[Function]}
      >
        <a
          className="inner-btn"
          title="Reblog"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M17,17H7V14L3,18L7,22V19H19V13H17M7,7H17V10L21,6L17,2V5H5V11H7V7Z"
              fill="currentColor"
            />
          </svg>
           
          
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`(2) Grid view 1`] = `
<div
  className="entry-list-item"
>
  <div
    className="item-header"
  >
    <div
      className="item-header-main"
    >
      <div
        className="author-part"
      >
        <a
          className="author-avatar"
          href="/@good-karma"
          onClick={[Function]}
        >
          <span
            className="user-avatar small"
            style={
              Object {
                "backgroundImage": "url(https://images.ecency.com/u/good-karma/avatar/small)",
              }
            }
          />
        </a>
        <a
          className="author notranslate"
          href="/@good-karma"
          onClick={[Function]}
        >
          good-karma
          <span
            className="author-reputation"
          >
            76
          </span>
        </a>
      </div>
      <a
        className="category"
        href="/hot/hive"
        onClick={[Function]}
      >
        hive
      </a>
      <span
        className="read-mark"
      />
      <span
        className="date"
        title="Monday, June 1, 2020 8:53 AM"
      >
        5 days
      </span>
    </div>
    <div
      className="item-header-features"
    />
  </div>
  <div
    className="item-body"
  >
    <div
      className="item-image"
    >
      <a
        href="/hive/@good-karma/awesome-hive"
        onClick={[Function]}
      >
        <img
          alt="Awesome Hive"
          onError={[Function]}
          src="https://images.ecency.com/p/HNWT6DgoBc1692QWn5trsGLxi44TrGuNZJMEdibBBbUKuho3EaW43JV4QzaQ258PEGNV2u43KAhU8r3Vjno4cBP8JeTFkbS2SCMjZZPprKzfMSdteRxSZxuPku4?format=match&mode=fit&width=600&height=500"
        />
      </a>
    </div>
    <div
      className="item-summary"
    >
      <a
        className="item-title"
        href="/hive/@good-karma/awesome-hive"
        onClick={[Function]}
      >
        Awesome Hive
      </a>
      <a
        className="item-body"
        href="/hive/@good-karma/awesome-hive"
        onClick={[Function]}
      >
        Hey developers, I just created Awesome-Hive, list of services and apps on Hive. If you don't know about Awesome project, take a look at here: Project contains list of services and links to various too
      </a>
    </div>
    <div
      className="item-controls"
    >
      <div
        className="entry-vote-btn"
        onClick={[Function]}
      >
        <div
          className="btn-vote btn-up-vote"
        >
          <span
            className="btn-inner"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="entry-payout  notranslate"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        $ 125.10
      </div>
      <div
        className="entry-votes notranslate"
      >
        <span
          className="inner-btn"
          onClick={[Function]}
          title="10 Votes"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M16 17V19H2V17S2 13 9 13 16 17 16 17M12.5 7.5A3.5 3.5 0 1 0 9 11A3.5 3.5 0 0 0 12.5 7.5M15.94 13A5.32 5.32 0 0 1 18 17V19H22V17S22 13.37 15.94 13M15 4A3.39 3.39 0 0 0 13.07 4.59A5 5 0 0 1 13.07 10.41A3.39 3.39 0 0 0 15 11A3.5 3.5 0 0 0 15 4Z"
              fill="currentColor"
            />
          </svg>
           
          10
        </span>
      </div>
      <a
        className="replies notranslate"
        href="/hive/@good-karma/awesome-hive"
        onClick={[Function]}
      >
        <span
          className="inner"
          title="31 responses. Click to respond."
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M18,8H6V6H18V8M18,11H6V9H18V11M18,14H6V12H18V14M22,4A2,2 0 0,0 20,2H4A2,2 0 0,0 2,4V16A2,2 0 0,0 4,18H18L22,22V4Z"
              fill="currentColor"
            />
          </svg>
           
          31
        </span>
      </a>
      <div
        className="entry-reblog-btn"
        onClick={[Function]}
      >
        <a
          className="inner-btn"
          title="Reblog"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M17,17H7V14L3,18L7,22V19H19V13H17M7,7H17V10L21,6L17,2V5H5V11H7V7Z"
              fill="currentColor"
            />
          </svg>
           
          
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`(3) Nsfw 1`] = `
<div
  className="entry-list-item"
>
  <div
    className="item-header"
  >
    <div
      className="item-header-main"
    >
      <div
        className="author-part"
      >
        <a
          className="author-avatar"
          href="/@good-karma"
          onClick={[Function]}
        >
          <span
            className="user-avatar small"
            style={
              Object {
                "backgroundImage": "url(https://images.ecency.com/u/good-karma/avatar/small)",
              }
            }
          />
        </a>
        <a
          className="author notranslate"
          href="/@good-karma"
          onClick={[Function]}
        >
          good-karma
          <span
            className="author-reputation"
          >
            76
          </span>
        </a>
      </div>
      <a
        className="category"
        href="/hot/hive"
        onClick={[Function]}
      >
        hive
      </a>
      <span
        className="read-mark"
      />
      <span
        className="date"
        title="Monday, June 1, 2020 8:53 AM"
      >
        5 days
      </span>
    </div>
    <div
      className="item-header-features"
    />
  </div>
  <div
    className="item-body"
  >
    <div
      className="item-image item-image-nsfw"
    >
      <img
        alt="Awesome Hive"
        src="nsfw.png"
      />
    </div>
    <div
      className="item-summary"
    >
      <div
        className="item-nsfw"
      >
        <span
          className="nsfw-badge"
        >
          NSFW
        </span>
      </div>
      <div
        className="item-nsfw-options"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Reveal this post
        </a>
         
         
        or
         
         
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "<a href=\\"/signup\\" class=\\"push-link\\">create an account</a> to save your preferences",
            }
          }
        />
        .
      </div>
    </div>
    <div
      className="item-controls"
    >
      <div
        className="entry-vote-btn"
        onClick={[Function]}
      >
        <div
          className="btn-vote btn-up-vote"
        >
          <span
            className="btn-inner"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="entry-payout  notranslate"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        $ 125.10
      </div>
      <div
        className="entry-votes notranslate"
      >
        <span
          className="inner-btn"
          onClick={[Function]}
          title="10 Votes"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M16 17V19H2V17S2 13 9 13 16 17 16 17M12.5 7.5A3.5 3.5 0 1 0 9 11A3.5 3.5 0 0 0 12.5 7.5M15.94 13A5.32 5.32 0 0 1 18 17V19H22V17S22 13.37 15.94 13M15 4A3.39 3.39 0 0 0 13.07 4.59A5 5 0 0 1 13.07 10.41A3.39 3.39 0 0 0 15 11A3.5 3.5 0 0 0 15 4Z"
              fill="currentColor"
            />
          </svg>
           
          10
        </span>
      </div>
      <a
        className="replies notranslate"
        href="/hive/@good-karma/awesome-hive"
        onClick={[Function]}
      >
        <span
          className="inner"
          title="31 responses. Click to respond."
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M18,8H6V6H18V8M18,11H6V9H18V11M18,14H6V12H18V14M22,4A2,2 0 0,0 20,2H4A2,2 0 0,0 2,4V16A2,2 0 0,0 4,18H18L22,22V4Z"
              fill="currentColor"
            />
          </svg>
           
          31
        </span>
      </a>
      <div
        className="entry-reblog-btn"
        onClick={[Function]}
      >
        <a
          className="inner-btn"
          title="Reblog"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M17,17H7V14L3,18L7,22V19H19V13H17M7,7H17V10L21,6L17,2V5H5V11H7V7Z"
              fill="currentColor"
            />
          </svg>
           
          
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`(4) Nsfw with active user 1`] = `
<div
  className="entry-list-item"
>
  <div
    className="item-header"
  >
    <div
      className="item-header-main"
    >
      <div
        className="author-part"
      >
        <a
          className="author-avatar"
          href="/@good-karma"
          onClick={[Function]}
        >
          <span
            className="user-avatar small"
            style={
              Object {
                "backgroundImage": "url(https://images.ecency.com/u/good-karma/avatar/small)",
              }
            }
          />
        </a>
        <a
          className="author notranslate"
          href="/@good-karma"
          onClick={[Function]}
        >
          good-karma
          <span
            className="author-reputation"
          >
            76
          </span>
        </a>
      </div>
      <a
        className="category"
        href="/hot/hive"
        onClick={[Function]}
      >
        hive
      </a>
      <span
        className="read-mark"
      />
      <span
        className="date"
        title="Monday, June 1, 2020 8:53 AM"
      >
        5 days
      </span>
    </div>
    <div
      className="item-header-features"
    />
  </div>
  <div
    className="item-body"
  >
    <div
      className="item-image item-image-nsfw"
    >
      <img
        alt="Awesome Hive"
        src="nsfw.png"
      />
    </div>
    <div
      className="item-summary"
    >
      <div
        className="item-nsfw"
      >
        <span
          className="nsfw-badge"
        >
          NSFW
        </span>
      </div>
      <div
        className="item-nsfw-options"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Reveal this post
        </a>
         
         
        or
         
         
        adjust your
         
        <a
          href="#"
          onClick={[Function]}
        >
          display preferences
        </a>
        .
      </div>
    </div>
    <div
      className="item-controls"
    >
      <div
        className="entry-vote-btn"
        onClick={[Function]}
      >
        <div
          className="btn-vote btn-up-vote"
        >
          <span
            className="btn-inner"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="entry-payout  notranslate"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        $ 125.10
      </div>
      <div
        className="entry-votes notranslate"
      >
        <span
          className="inner-btn"
          onClick={[Function]}
          title="10 Votes"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M16 17V19H2V17S2 13 9 13 16 17 16 17M12.5 7.5A3.5 3.5 0 1 0 9 11A3.5 3.5 0 0 0 12.5 7.5M15.94 13A5.32 5.32 0 0 1 18 17V19H22V17S22 13.37 15.94 13M15 4A3.39 3.39 0 0 0 13.07 4.59A5 5 0 0 1 13.07 10.41A3.39 3.39 0 0 0 15 11A3.5 3.5 0 0 0 15 4Z"
              fill="currentColor"
            />
          </svg>
           
          10
        </span>
      </div>
      <a
        className="replies notranslate"
        href="/hive/@good-karma/awesome-hive"
        onClick={[Function]}
      >
        <span
          className="inner"
          title="31 responses. Click to respond."
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M18,8H6V6H18V8M18,11H6V9H18V11M18,14H6V12H18V14M22,4A2,2 0 0,0 20,2H4A2,2 0 0,0 2,4V16A2,2 0 0,0 4,18H18L22,22V4Z"
              fill="currentColor"
            />
          </svg>
           
          31
        </span>
      </a>
      <div
        className="entry-reblog-btn"
        onClick={[Function]}
      >
        <a
          className="inner-btn"
          title="Reblog"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M17,17H7V14L3,18L7,22V19H19V13H17M7,7H17V10L21,6L17,2V5H5V11H7V7Z"
              fill="currentColor"
            />
          </svg>
           
          
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`(5) Nsfw but allowed 1`] = `
<div
  className="entry-list-item"
>
  <div
    className="item-header"
  >
    <div
      className="item-header-main"
    >
      <div
        className="author-part"
      >
        <a
          className="author-avatar"
          href="/@good-karma"
          onClick={[Function]}
        >
          <span
            className="user-avatar small"
            style={
              Object {
                "backgroundImage": "url(https://images.ecency.com/u/good-karma/avatar/small)",
              }
            }
          />
        </a>
        <a
          className="author notranslate"
          href="/@good-karma"
          onClick={[Function]}
        >
          good-karma
          <span
            className="author-reputation"
          >
            76
          </span>
        </a>
      </div>
      <a
        className="category"
        href="/hot/hive"
        onClick={[Function]}
      >
        hive
      </a>
      <span
        className="read-mark"
      />
      <span
        className="date"
        title="Monday, June 1, 2020 8:53 AM"
      >
        5 days
      </span>
    </div>
    <div
      className="item-header-features"
    />
  </div>
  <div
    className="item-body"
  >
    <div
      className="item-image"
    >
      <a
        href="/hive/@good-karma/awesome-hive"
        onClick={[Function]}
      >
        <picture>
          <source
            media="(min-width: 576px)"
            srcSet="https://images.ecency.com/p/HNWT6DgoBc1692QWn5trsGLxi44TrGuNZJMEdibBBbUKuho3EaW43JV4QzaQ258PEGNV2u43KAhU8r3Vjno4cBP8JeTFkbS2SCMjZZPprKzfMSdteRxSZxuPku4?format=match&mode=fit&width=260&height=200"
          />
          <img
            alt="Awesome Hive"
            onError={[Function]}
            srcSet="https://images.ecency.com/p/HNWT6DgoBc1692QWn5trsGLxi44TrGuNZJMEdibBBbUKuho3EaW43JV4QzaQ258PEGNV2u43KAhU8r3Vjno4cBP8JeTFkbS2SCMjZZPprKzfMSdteRxSZxuPku4?format=match&mode=fit&width=600&height=500"
          />
        </picture>
      </a>
    </div>
    <div
      className="item-summary"
    >
      <a
        className="item-title"
        href="/hive/@good-karma/awesome-hive"
        onClick={[Function]}
      >
        Awesome Hive
      </a>
      <a
        className="item-body"
        href="/hive/@good-karma/awesome-hive"
        onClick={[Function]}
      >
        Hey developers, I just created Awesome-Hive, list of services and apps on Hive. If you don't know about Awesome project, take a look at here: Project contains list of services and links to various too
      </a>
    </div>
    <div
      className="item-controls"
    >
      <div
        className="entry-vote-btn"
        onClick={[Function]}
      >
        <div
          className="btn-vote btn-up-vote"
        >
          <span
            className="btn-inner"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="entry-payout  notranslate"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        $ 125.10
      </div>
      <div
        className="entry-votes notranslate"
      >
        <span
          className="inner-btn"
          onClick={[Function]}
          title="10 Votes"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M16 17V19H2V17S2 13 9 13 16 17 16 17M12.5 7.5A3.5 3.5 0 1 0 9 11A3.5 3.5 0 0 0 12.5 7.5M15.94 13A5.32 5.32 0 0 1 18 17V19H22V17S22 13.37 15.94 13M15 4A3.39 3.39 0 0 0 13.07 4.59A5 5 0 0 1 13.07 10.41A3.39 3.39 0 0 0 15 11A3.5 3.5 0 0 0 15 4Z"
              fill="currentColor"
            />
          </svg>
           
          10
        </span>
      </div>
      <a
        className="replies notranslate"
        href="/hive/@good-karma/awesome-hive"
        onClick={[Function]}
      >
        <span
          className="inner"
          title="31 responses. Click to respond."
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M18,8H6V6H18V8M18,11H6V9H18V11M18,14H6V12H18V14M22,4A2,2 0 0,0 20,2H4A2,2 0 0,0 2,4V16A2,2 0 0,0 4,18H18L22,22V4Z"
              fill="currentColor"
            />
          </svg>
           
          31
        </span>
      </a>
      <div
        className="entry-reblog-btn"
        onClick={[Function]}
      >
        <a
          className="inner-btn"
          title="Reblog"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M17,17H7V14L3,18L7,22V19H19V13H17M7,7H17V10L21,6L17,2V5H5V11H7V7Z"
              fill="currentColor"
            />
          </svg>
           
          
        </a>
      </div>
    </div>
  </div>
</div>
`;
